### PR TITLE
[Merged by Bors] - feat: More simple simple graph lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
A bunch of simple lemmas for simple graphs.

---

From LeanCamCombi

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

File: Mathlib/Combinatorics/SimpleGraph/Basic.lean [hunk 1]
```diff
@@ -1745,13 +1825,27 @@ infixl:50 " ≃g " => Iso
...
+@[simp, norm_cast] lemma coe_id : ⇑(Hom.id : G →g G) = _root_.id := rfl
+
+instance [Subsingleton (V → W)] : Subsingleton (G →g H) :=
+  FunLike.coe_injective.subsingleton
+
+instance [IsEmpty V] : Unique (G →g H) where
+  default := ⟨isEmptyElim, fun {a} ↦ isEmptyElim a⟩
+  uniq _ := Subsingleton.elim _ _
+
+noncomputable instance [Fintype V] [Fintype W] : Fintype (G →g H) := by
```
Comment (kmill) [inline Mathlib/Combinatorics/SimpleGraph/Basic.lean:1844]: We shouldn't have noncomputable Fintype instances -- anyone who needs one should introduce a local instance (for example from the `Finite` instance below).

If I'm not mistaken, `[DecidableRel G.Adj]` and `[DecidableRel H.Adj]` suffice to make this computable. You might need `[DecidableEq V]` too if you want to go through `Pi.Fintype` and filter the result.

File: Mathlib/Logic/Relation.lean [hunk 2]
```diff
@@ -47,7 +48,7 @@ the bundled version, see `Rel`.
 
 open Function
 
-variable {α β γ δ : Type*}
+variable {α β γ δ ε ν : Type*}
```
Comment (kmill) [inline Mathlib/Logic/Relation.lean:51]: Isn't the letter after epsilon zeta, not nu?

File: Mathlib/Combinatorics/SimpleGraph/Basic.lean [hunk 3]
```diff
@@ -693,6 +715,14 @@ theorem fromEdgeSet_mono {s t : Set (Sym2 V)} (h : s ⊆ t) : fromEdgeSet s ≤
   exact fun vws _ => h vws
 #align simple_graph.from_edge_set_mono SimpleGraph.fromEdgeSet_mono
 
+@[simp] lemma disjoint_fromEdgeSet : Disjoint G (fromEdgeSet s) ↔ Disjoint G.edgeSet s := by
+  conv_rhs => rw [←Set.diff_union_inter s {e : Sym2 V | e.IsDiag}]
+  rw [←disjoint_edgeSet,  edgeSet_fromEdgeSet, Set.disjoint_union_right, and_iff_left]
+  exact Set.disjoint_left.2 fun e he he' ↦ not_isDiag_of_mem_edgeSet _ he he'.2
```
[the remark this answers is not in the document]
Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Basic.lean:721]: I don't really mind which way the lemma goes, but one way should be simp.

File: Mathlib/Combinatorics/SimpleGraph/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Combinatorics/SimpleGraph/Basic.lean:1844]: Well technically there are cases where no computable fintype instance can be provided. But I guess this is not the case here.

Comment (kmill) [reply Mathlib/Combinatorics/SimpleGraph/Basic.lean:1844]: In cases where we can't provide a computable fintype instance, I don't think we should provide a fintype instance at all (at best, we could have a non-instance `def`). If we're willing to have non-computable Fintype instances, we may as well stop worrying about decidability and switch everything to Finite.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
This should now have been fixed on master.

bors merge

**Comment by YaelDillies** [comment]
bors merge

**Comment by eric-wieser** [comment]
Let's remove this test in a standalone PR

bors r-

**Comment by YaelDillies** [comment]
Finally the time, hopefully!

bors merge

**Comment by YaelDillies** [comment]
Famous last words.

bors merge

**Pull request closed without merging**